first_order_model/shrink_util.py

```diff
@@ -0,0 +1,1266 @@
+from tqdm import trange
```

Comment:
Please do the following for this file:
1) add function comments, and block comments to all the code in here.
2) clean up any commented out code
3) ensure lines are no longer than 80 characters
4) special names such as layer names are defined separately in a file that you can export it from or just used from a function output that allows you to filter which names you want.
5) Use consistent spacing, such as no line between function name and comment/code, 2 lines between different functions. Space between operands and operators.
6) Remove print statements.

Will review the file only after that. Worth looking at whether you can also split this into two files: helpers and core logic